### ISSUE TITLE
🐛 [BUG]: Error when calling fitView and nodes haven't initialized

### ISSUE THREAD

**Issue by nBukaty** [bcakmakoglu/vue-flow#1009]
### Is there an existing issue for this?

- [X] I have searched the existing issues and this is a new bug.

### Current Behavior

After loading nodes/edges on mount, then shortly after removing and loading new nodes/edges, the canvas stops working. It appears this happens when the new nodes have not been initialized when fitView is executing. The error "Error: <pattern> attribute y: Expected length, "NaN". " shows up repeatedly in the console.
![image](https://github.com/bcakmakoglu/vue-flow/assets/19231471/5b86ee78-9ee2-4d24-9d80-95176c324bd7)

### Expected Behavior

The background should remain responsive. 

### Steps To Reproduce

I originally ran into this when loading data from a store, then shortly after loading new data from an API. I used the examples section of this project to recreate the issue. 
```vue
<script  lang="ts" setup>
import type { Node, Styles } from '@vue-flow/core'
import { useVueFlow } from '@vue-flow/core'
import { Background } from '@vue-flow/background'
import { Controls } from '@vue-flow/controls'
import { MiniMap } from '@vue-flow/minimap'
import { onMounted, watch } from 'vue';

const vueFlow = useVueFlow();
const elements = ref<number[]>([1,2,3])

onMounted(async () => {
  // trigger the watcher
  elements.value.pop();
  vueFlow.addNodes([{
    id: '1',
    type: 'input',
    label: 'Welcome to <strong>Vue VueFlow!</strong>',
    position: {x: 250, y: 0},
  },
    {
      id: '2',
      label: 'This is a <strong>default node</strong>',
      position: {x: 100, y: 100},
    }]);
  vueFlow.addEdges([{id: 'e1-2', source: '1', target: '2', label: 'this is an edge label'}]);

  vueFlow.fitView();
})
watch(() => elements.value.length, (value, oldValue) => {
  if (value < oldValue) {
    // Seems to happen at a specific time in the lifecycle so adding this delay
    setTimeout(() => {
      vueFlow.removeNodes(['1', '2'])
      vueFlow.removeEdges(['e1-2'])
      vueFlow.addNodes([{
        id: '12',
        type: 'input',
        label: 'Example',
        position: {x: 250, y: 0},
      },
        {
          id: '13',
          type: 'input',
          label: 'Example',
          position: {x: 250, y: 50},
        }]);
      vueFlow.addEdges({id: 'e12-13', source: '12', target: '13', label: 'this is an edge label'})
      vueFlow.fitView();
    }, 10);

  }
})
</script>
<template>
  <VueFlow>
    <MiniMap :node-border-radius="2"/>
    <Controls/>
    <Background color="#aaa" :gap="20"/>
  </VueFlow>
</template>
```

**Comment by bcakmakoglu** [comment]
Thanks for reporting but my question for this case would be: Why are you even doing this? What possible use-case is there for this? ^^

**Comment by nBukaty** [comment]
The use case would be that you use a store or something that is persistent for your data. If you don't clear that store before mounting vue-flow then you could run into this issue. For example, the user could open a page that previously displayed data that is still in the store. Opening that page again with some new params would cause new data to be loaded into the store. This would cause the initial data to be loaded followed by the new data and vue-flow would freeze. My workaround for this is to conditionally clear the store when displaying something different. If nothing else, this page can act as a solution for anyone else that encounters this.

**Comment by bcakmakoglu** [comment]
mh... okay i get that 👍 
I'll see what I can do :)

**Comment by bcakmakoglu** [comment]
Should be fixed with `1.21.2` - if the issue persists, pleas re-open the issue
